common/djangoapps/student/forms.py

```diff
@@ -396,16 +394,29 @@ def clean_email(self):
             )
         return email
 
-    def clean_year_of_birth(self):
-        """
-        Parse year_of_birth to an integer, but just use None instead of raising
-        an error if it is malformed
-        """
-        try:
-            year_str = self.cleaned_data["year_of_birth"]
-            return int(year_str) if year_str is not None else None
-        except ValueError:
-            return None
+    def clean_org_name(self):
+        """ Enforce organization related field conditions"""
+        clean_org_name = self.cleaned_data.get("org_name")
+        clean_org_size = self.cleaned_data.get("org_size")
+        clean_org_type = self.cleaned_data.get("org_type")
+
+        # User is affiliated with some organization
+        if clean_org_name:
+            # Check if organization already exists and does have a size populated
+            existing_org = Organization.objects.filter(label=clean_org_name).first()
+
+            if existing_org:
+                existing_org_size = existing_org.total_employees
+                if not existing_org_size and not clean_org_size:
+                    raise forms.ValidationError(_("Organization size not provided for org with unpopulated size"))
+                elif existing_org_size and clean_org_size:
+                    raise forms.ValidationError(_("Organization size provided for existing org with populated size"))
+            else:
+                if not clean_org_size or not clean_org_type:
```

Comment:
Can we have those check above to avoid query if user hasn't provided  `clean_org_size` or `clean_org_type`

Reply:
1. User can give none of `org_name`, `org_type`, `org_size` which would mean unaffiliated
2. If user gives `org_name`, we first have to check if it is an existing name. If the org already exists, it's possible that `org_size` for that org is not populated, in which case we also need the user to give `org_size`. So we have to do the query in order to see if `org_size` is required or not.

Reply:
That clarifies...